packages/components/src/alignment-matrix-control/icon.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { ALIGNMENTS, getAlignmentIndex } from './utils';
+import {
+	Root,
+	Cell,
+	Point,
+} from './styles/alignment-matrix-control-icon-styles';
+
+const BASE_SIZE = 24;
+
+export default function AlignmentMatrixControlIcon( {
+	className,
+	disablePointerEvents = true,
+	height,
+	size: sizeProp = BASE_SIZE,
+	style = {},
+	value = 'center',
+	width,
+	...props
+} ) {
+	const alignIndex = getAlignmentIndex( value );
+	const size = width || height || sizeProp;
+	const scale = ( size / BASE_SIZE ).toFixed( 2 );
+
+	const classes = classnames(
+		'component-alignment-matrix-control-icon',
+		className
+	);
+
+	const styles = {
+		...style,
+		transform: `scale(${ scale })`,
+	};
+
+	return (
+		<Root
+			{ ...props }
+			className={ classes }
+			disablePointerEvents={ disablePointerEvents }
+			role="presentation"
+			size={ size }
+			style={ styles }
+		>
+			{ ALIGNMENTS.map( ( align, index ) => {
+				const isActive = alignIndex === index;
```

Comment:
Why not just compare the current `align` to `value`?

Reply:
The `value` coming in may not match `align`.
The `getAlignmentIndex` function parses/transforms a `value` prop for accurate reading and comparison purposes.

Originally, the transforming part was more complicated as it allowed for more flexibility in positions, example: both `top center` and `center top` evaluated to the same thing.

Since working with @diegohaz , that logic has been simplified.
It now handles the ability to read `center` as `center center`.